src/Workspaces/CSharp/Portable/SemanticModelReuse/CSharpSemanticModelReuseLanguageService.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SemanticModelReuse;
+
+namespace Microsoft.CodeAnalysis.CSharp.SemanticModelReuse
+{
+    [ExportLanguageService(typeof(ISemanticModelReuseLanguageService), LanguageNames.CSharp), Shared]
+    internal class CSharpSemanticModelReuseLanguageService : ISemanticModelReuseLanguageService
```

Comment:
note: there is a workspace-service and a lang-service.  teh former is what is used for the basic semantic-model caching.  The latter is waht it used for doing vb/c# specific work in that process.